source/margarita/upgrading.rst

```diff
@@ -0,0 +1,557 @@
+      +++ b/requirements/base.txt
+      +
+      +django-dotenv==1.3.0
+
+#. Modify ``README.rst`` (and follow those instructions for your local setup):
+
+   .. code-block:: diff
+
+      Modified   README.rst
+      diff --git a/README.rst b/README.rst
+      index c45b564..8c456be 100644
+      --- a/README.rst
+      +++ b/README.rst
+      @@ -27,7 +27,9 @@ necessary requirements::
+
+      -Then create a local settings file and set your ``DJANGO_SETTINGS_MODULE`` to use it::
+      +Next, we'll set up our local environment variables. We use `django-dotenv
+      +<https://github.com/jpadilla/django-dotenv>`_ to help with this. It reads environment variables
+      +located in a file name ``.env`` in the top level directory of the project. The only variable we need
+      +to start is ``DJANGO_SETTINGS_MODULE``::
+
+      -    cp myproject/settings/local.example.py myproject/settings/local_dev.py
+      -    echo "export DJANGO_SETTINGS_MODULE=myproject.settings.local_dev" >> $VIRTUAL_ENV/bin/postactivate
+      -    echo "unset DJANGO_SETTINGS_MODULE" >> $VIRTUAL_ENV/bin/postdeactivate
+      +    (myproject)$ cp myproject/settings/local.example.py myproject/settings/local.py
+      +    (myproject)$ echo "DJANGO_SETTINGS_MODULE=myproject.settings.local" > .env
+
+
+Update ALLOWED_HOSTS
+--------------------
+
+Find the ``ALLOWED_HOSTS`` setting (probably in ``staging.py``) and change it to use ``DOMAIN``:
+
+.. code-block:: diff
+
+   Modified   myproject/settings/staging.py
+   diff --git a/myproject/settings/staging.py b/myproject/settings/staging.py
+   index db7b3b4..be4024d 100644
+   --- a/myproject/settings/staging.py
+   +++ b/myproject/settings/staging.py
+   @@ -34,3 +34,3 @@ SESSION_COOKIE_HTTPONLY = True
+
+   -ALLOWED_HOSTS = os.environ.get('ALLOWED_HOSTS', '').split(';')
+   +ALLOWED_HOSTS = [os.environ['DOMAIN']]
+
+Frontend Improvements
+---------------------
+
+Prepare for Calvin's frontend improvements. Add a *dummy* ``package.json`` which can be updated
+later. Until it is updated, the frontend improvements won't take effect:
+
+.. code-block:: json
+
+   {
+     "name": "",
+     "version": "0.0.0",
+     "description": "",
+     "main": "",
+     "engines" : {
+       "node" : ">=4.2 <4.3"
+     },
+     "scripts": {
+       "build": "true"
+     },
+     "author": "",
+     "license": "",
+     "dependencies": {},
+     "devDependencies": {}
+   }
+
+
+Miscellaneous work
+------------------
+
+Technically, you can skip the steps in this section and come back to them later. Even without them,
+you should be able to get a server upgraded, but they **will** have to be done at some point.
+
+1. Port any useful functions in ``fabfile.py.old`` to the new fabfile, then remove the old one.
+
+#. Get a copy of the ``Makefile`` from the project template, porting any functions in your existing
+   one to the new one, if needed.
+
+#. Review everything in ``salt.old`` to see which pieces are specific to your project and need to
```

Comment:
If we preserved any custom states above, then we can simplify this slightly to just look to see what states remain in the project's salt directory and see if they can be moved to Margarita.